src/views/bookmark_x/functional_types.ts

```diff
@@ -379,6 +380,10 @@ export class ViewItemUriMap extends UriMap<BookmarkTreeItem> {
         this.keys().forEach(key => {
             // reset icon status
             let tvi = this.get(key);
+            if(SpaceSortItem.sorting_item === tvi){
+                // item is sorting, use sorting icon
+                return;
+            }
```

Comment:
I think the tvi icon updateing here can be rewrite, some logic can be merged into a class like e.g. `TviIconManager` 

call it's method and refresh the icon.

there may be some difficulty to get infos from controller.